bluemira/utilities/prominence.py

```diff
@@ -0,0 +1,127 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""
+Prominence API
+"""
+
+import sys
+from pathlib import Path
+from unittest.mock import patch
+
+from bluemira.base.look_and_feel import bluemira_debug, bluemira_print, bluemira_warn
+from bluemira.utilities.tools import get_module
+
+
+class ProminenceDownloader:
+    """
+    Object to import prominence's binary file to enable job data downloading.
+
+    the binary file has a lot of python code directly in it that is
+    not accessible from the prominence module.
```

Comment:
If the code is not accessible from the module, is there any guarantee that its API is stable? I'm not familiar with `prominence` so there may not be, but is there no way to achieve this functionality through some public API? Be that through calling the executable in a subprocess or otherwise?

Reply:
No I cant guarantee any of that. There are two cli programs that are in the organisation https://github.com/prominence-eosc but the pinned one (which hasnt been updated in a while) isnt v1 yet. The other cli system doesnt seem to have any download functionality.

prominence (as far as I'm aware) is used by the likes of JETTO to submit jobs but may in the future be useful when we start doing parameter scans.

We could call it through subprocess but I felt it was counter intuitive to go through bash back to python especially when you may have to reauthorise on each command when this is expanded into a full API. We probably should discuss further.